plugins/batterystat.koplugin/main.lua

```diff
@@ -207,10 +202,47 @@ function BatteryStat:showStatistics()
     table.insert(kv_pairs, "----------")
     table.insert(kv_pairs, {_("Historical records are dumped to"), ""})
     table.insert(kv_pairs, {self.dump_file, ""})
-    UIManager:show(KeyValuePage:new{
+    table.insert(kv_pairs, "----------")
+    table.insert(kv_pairs, {_("Should you like to reset the data,"), "",
+                            callback = function()
+                                self:resetAll()
+                                self:restart()
+                            end})
+    table.insert(kv_pairs, {_("please tap here."), "",
+                            callback = function()
+                                self:resetAll()
+                                self:restart()
+                            end})
+    self.kv_page = KeyValuePage:new{
         title = _("Battery statistics"),
         kv_pairs = kv_pairs,
-    })
+    }
+    UIManager:show(self.kv_page)
+end
+
+function BatteryStat:reset(withCharging, withDischarging)
+    self:dumpToText()
+    self.awake = Usage:new()
+    self.sleeping = Usage:new()
+
+    if withCharging then
+        self.charging = Usage:new()
+    end
+    if withDischarging then
+        self.discharging = Usage:new()
+    end
+end
+
+function BatteryStat:resetAll()
+    self:reset(true, true)
+    self.charging_state = State:new()
+    self.awake_state = State:new()
+end
+
+function BatteryStat:restart()
+    assert(self.kv_page ~= nil)
```

Comment:
Yes, here is the tradeoff, `dbg:guard` will get optimized to nothing in non-debug mode. But `dbg:dassert` will still introduce runtime penalties because the boolean statements are still evaluated every time before getting passed to `dassert`. Keep in mind that LUAJIT really doesn't like branching ;)

Another issue with dassert is it will still crash the program if the boolean statement has a bug. For example, when `p1 == nil`.

In languages like C++/C, you can define dassert as a macro and have the compiler optimize it to nothing at compile time. Unfortunately, Lua doesn't have macro support.

So yes, there are reasons for why it looks weird :)

If you care about performance and safety, it's better to reimplement all the busted like helper methods like @Frenzie mentioned. i.e. do the following:

```lua
dbg:dassert_is_gt(p1, 0)
```
instead of

```lua
dbg:dassert(p1 > 0)
```

Then you can set `dbg:dassert_is_gt` to a noop (empty function) at module load time. The drawback is you will need to write more code ;)